docs/madr/decisions/076-naming-internal-envoy-resources.md

```diff
@@ -0,0 +1,319 @@
+#### 1. User resources
+
+Names for these resources are defined in [Defining and migrating to consistent naming for non-system Envoy resources and stats](./077-migrating-to-consistent-and-well-defined-naming-for-non-system-envoy-resources-and-stats.md).
+
+#### 2. System resources that can be tracked back to user resource
+
+When possible, use `system_<kri>`. It should be justified when designing why a user resource is system and not just `<kri>`.
+For example, when a resource isn't part of regular service-to-service traffic path, is shared across the system (like Kong Mesh's `MeshGlobalRateLimit` service), or simply needs to be easy to exclude from metrics collection.
+
+There are other cases where a resource is the result of merging multiple resources together (MeshTrafficPermission RBAC filters, MeshPassthrough, etc.).
+This is out of scope and requires a separate MADR.
+
+#### 3. System resources that are not related to user resource
+
+When adding such resource the name should be explicit enough to help anyone understand where this is coming from.
+The name should be namespaced so that it groups similar resources together and avoids collisions with other system resources.
+
+For example dynamic config fetcher creates one listener and one route per object type.
+We will namespace it as `system_dynamicconfig` and the DNS route will be `system_dynamicconfig_dns` and MeshMetric route will be `system_dynamicconfig_meshmetric`.
+
+If any related configuration option or annotation name exists it should be taken into account.
+
+#### 4. Resource contextual to the Dataplane
+
+These resources are defined in [Defining and migrating to consistent naming for non-system Envoy resources and stats](./077-migrating-to-consistent-and-well-defined-naming-for-non-system-envoy-resources-and-stats.md).
+
+#### 5. Secrets
+
+There is nothing special about secrets, they adhere to the same rules (1-4) and can have exceptions (6).
+
+#### 6. Exceptions
+
+We can't control the name of external resources (i.e. SPIRE SDS secrets).
+Any new exception and case should result in a MADR which can be linked from here.
+
+#### 7. Regex to match system names
+
+For `system_<kri>` resources we will use the same regex as used for KRI but prefixed with `system_`.
+
+For other names we will use `^system_([a-z0-9-]*_?)+$`, here is an example of the usage: https://regex101.com/r/Ic1bk5/2.
+
+### Currently existing system resources and their new naming
+
+| Resource type | Current Name                                                 | New Name                        | Example                                                      |
+|---------------|--------------------------------------------------------------|---------------------------------|--------------------------------------------------------------|
+| Cluster       | kuma:envoy:admin                                             | system_envoy_admin              |                                                              | 
```

Comment:
do we need `_envoy_`? maybe just `system_admin`?

Reply:
`envoy` is kind of a "namespace" which allows us to expand it in the future and using `system_admin` also blocks us from using `admin` as a "namespace" if we would like to use it